src/main/java/core/model/match/MatchKurzInfo.java

```diff
@@ -553,4 +556,73 @@ public String getResultLong() {
 		}
 		return getMatchdetails().getResultLong();
 	}
+
+	public double getExperienceIncrease(int minutes) {
+		double p;
+		var _matchType = this.getMatchTypeExtended();
+		if (_matchType instanceof MatchType){
+			p = switch ((MatchType) _matchType){
+				case CUP, QUALIFICATION -> 2d;
+				case FRIENDLYNORMAL, FRIENDLYCUPRULES -> .1;
+				case INTFRIENDLYCUPRULES, INTFRIENDLYNORMAL -> .2;
+				case NATIONALFRIENDLY, LEAGUE -> 1d;
+				case MASTERS -> 5d;
+				case NATIONALCOMPCUPRULES -> 16d;
+				case NATIONALCOMPNORMAL -> 8d;
+				case TOURNAMENTGROUP -> getTournamentExperienceFactor(TournamentType.getById(this.iMatchContextId), false);
+				case TOURNAMENTPLAYOFF -> getTournamentExperienceFactor(TournamentType.getById(this.iMatchContextId), true);
+				default -> 0d;
+			};
+		} else{
+			//case MatchTypeExtended.EMERALDCUP, MatchTypeExtended.RUBYCUP, MatchTypeExtended.SAPPHIRECUP -> .5;
+			p = 0.5;
+		}
+		return minutes * p / 90. / 28.571;
+	}
+
+	/**
+	 * get experience factor of tournament natianal team matches
+	 * TODO: check if playoffs are really tagged by MatchType.TOURNAMENTPLAYOFF
+	 * @param tournamentType tournament type of the match, null if no tournament match
+	 * @param isPlayOff boolean, true if play off tournament match
+	 * @return factor of league match experience increases (3.5)
+	 */
+	private double getTournamentExperienceFactor(@Nullable TournamentType tournamentType, boolean isPlayOff) {
+		if ( tournamentType == null) return 0d;
+		if ( tournamentType.isWorldCup()){
+			// WM-Spiel	28
```

Comment:
Would you be able to have those comments in English? Or remove them if you don't think they add value. ;-)

Reply:
translation found - changed